classes/class-wc-connect-error-notice.php

```diff
@@ -53,10 +53,23 @@ private function show_notice() {
 			$link_status = admin_url( 'admin.php?page=wc-status&tab=connect' );
 			$link_dismiss = add_query_arg( array( 'wc-connect-error-notice' => 'disable' ) );
 
-			$message = sprintf(
-				__( 'An error occurred in WooCommerce Services. Details are logged <a href="%s">here</a>.', 'woocommerce-services' ),
-				$link_status, $link_dismiss
-			);
+			$error = $this->notice_enabled();
+			if ( is_wp_error( $error ) && 'product_missing_weight' === $error->get_error_code() ) {
+				$matches = array();
+				preg_match( '/ID:\s([0-9]+)/', $error->get_error_message(), $matches );
```

Comment:
Can we avoid the regexp by adding data to the `WP_Error` here: https://github.com/Automattic/woocommerce-services/blob/master/classes/class-wc-connect-api-client.php#L97 ?

Reply:
This was fixed in https://github.com/Automattic/woocommerce-services/commit/dfa10e4601a22c9d6e2190e3b84fc7df45045956